src/Workspaces/CoreTest/Execution/SnapshotSerializationTests.cs

```diff
@@ -493,6 +495,23 @@ public class SnapshotSerializationTests : SnapshotSerializationTestBase
             }
         }
 
+        [Fact]
+        public async Task UnknownLanguageTest()
+        {
+            var hostServices = MefHostServices.Create(MefHostServices.DefaultAssemblies.Add(typeof(NullLanguageService).Assembly));
+
+            var solution = new AdhocWorkspace(hostServices).CurrentSolution;
+
+            var project1 = solution.AddProject("Project", "Project.dll", NullLanguageService.TestLanguage);
```

Comment:
I believe we already have a language for this.  We have tests for this. look for NoCompilation